### PR TITLE
Master add noninteractive mode

### PR DESCRIPTION
**Pull request by jkonecny12**

### CODE REVIEW

File: pyanaconda/anaconda_argparse.py [hunk 1]
```diff
@@ -383,13 +385,20 @@ def getArgumentParser(version_string, boot_cmdline=None):
     # Version
     ap.add_argument('--version', action='version', version="%(prog)s " + version_string)
 
+    class ParseCmdlineMode(Action):
```
Comment (M4rtinK) [inline pyanaconda/anaconda_argparse.py]: I would suggest `SetCmdlineMode` (or something similar) as the Action subclass actually doesn't do any parsing in this case.

File: pyanaconda/ui/gui/hubs/__init__.py [hunk 2]
```diff
@@ -352,7 +361,14 @@ def refresh(self):
         GUIObject.refresh(self)
         self._createBox()
 
-        GLib.timeout_add(100, self._update_spokes)
+        for hub in Hub._hubs_collection:
+            if hub.timeout_id is not None:
+                log.debug("Disable event loop for hub %s", hub.__class__.__name__)
```
Comment (M4rtinK) [inline pyanaconda/ui/gui/hubs/__init__.py]: Just a nitpick, but I would change this to "Disabling event loop for hub %s". :)

File: pyanaconda/ui/gui/hubs/__init__.py [hunk 3]
```diff
@@ -352,7 +361,14 @@ def refresh(self):
...
+                GLib.source_remove(hub.timeout_id)
+                hub.timeout_id = None
+
+        log.debug("Start event loop for hub %s", self.__class__.__name__)
```
Comment (M4rtinK) [inline pyanaconda/ui/gui/hubs/__init__.py]: Similar nitpick/suggesting - "Starting event loop for hub %s"

**Comment by jkonecny12** [comment]
**Updated**: tests were fixed

**Comment by jkonecny12** [comment]
**Updated**: Martin's suggestions should be implemented.

File: anaconda.py [hunk 4]
```diff
@@ -603,10 +603,10 @@ def _earlyExceptionHandler(ty, value, traceback):
     from pykickstart import constants as pykickstart_constants
```
Comment (vpodzime) [inline anaconda.py]: "*without a hacks*" -> "*without hacks*" in the commit message.

Comment (vpodzime) [inline anaconda.py]: Also "*Only TEXT and GUI mode is now used*" "*Only TEXT and GUI modes are now used*"

File: pyanaconda/constants.py [hunk 5]
```diff
@@ -201,21 +201,11 @@
 NTP_SERVER_NOK = 1
 NTP_SERVER_QUERY = 2
 
-DISPLAY_MODE_GUI = "g"
-DISPLAY_MODE_TUI = "t"
-DISPLAY_MODE_NONINTERACTIVE_TUI = "c"
+# Display modes
+DISPLAY_MODE_GUI = 0
+DISPLAY_MODE_TUI = 1
```
Comment (vpodzime) [inline pyanaconda/constants.py]: Just a side note while looking at this -- maybe it's time to move to ``Enum``s in Anaconda?

File: pyanaconda/errors.py [hunk 6]
```diff
@@ -20,7 +20,7 @@
```
Comment (vpodzime) [inline pyanaconda/errors.py]: "*If this new exception is caught, the installation will be killed.*"

File: pyanaconda/ui/gui/hubs/__init__.py [hunk 7]
```diff
@@ -60,6 +60,7 @@ class Hub(GUIObject, common.Hub):
     """
```
Comment (vpodzime) [inline pyanaconda/ui/gui/hubs/__init__.py]: "*when a new Hub is created, it will first disable...and then it will enable...*"

Comment (vpodzime) [inline pyanaconda/ui/gui/hubs/__init__.py]: Also, the loops are disabled/enabled when the hub is show, not when it is created, AFAICT.

File: pyanaconda/ui/gui/spokes/password.py [hunk 8]
```diff
@@ -27,6 +27,7 @@
 from pyanaconda.ui.gui.utils import set_password_visibility
```
Comment (vpodzime) [inline pyanaconda/ui/gui/spokes/password.py]: The commit message seems to be unrelated to the actual changes -- more details needed.

File: pyanaconda/errors.py [hunk 9]
```diff
@@ -300,6 +301,9 @@ def cb(self, exn):
         if not self.ui:
             raise exn
 
+        if not flags.ksprompt:
+            raise NonInteractiveError("Non interactive installation failed: %s" % exn)
+
```
Comment (vpodzime) [inline pyanaconda/errors.py]: A very specific commit message for such generic change.

File: docs/boot-options.rst [hunk 10]
```diff
@@ -380,6 +380,16 @@ inst.text
 Run the installer using a limited text-based UI. Unless you're using a
 kickstart file this probably isn't a good idea; you should use VNC instead.
 
+.. inst.noninteractive
+
+inst.noninteractive
+^^^^^^^^^^^^^^^^^^^
+
+Run the installer in non interactive mode. This mode does not allow any user
```
Comment (vpodzime) [inline docs/boot-options.rst]: "*in a non-interactive...*"

File: docs/boot-options.rst [hunk 11]
```diff
@@ -380,6 +380,16 @@ inst.text
...
+interaction and can be used with graphical or text mode. With text mode it
+will behave the same as ``inst.cmdline`` mode.
```
Comment (vpodzime) [inline docs/boot-options.rst]: "*behaves the same as the...*"

File: docs/boot-options.rst [hunk 12]
```diff
@@ -380,6 +380,16 @@ inst.text
...
+All options must be specified in a kickstart file or on the command line.
```
Comment (vpodzime) [inline docs/boot-options.rst]: What does this sentence mean? It's quite confusing even for me who know how these new things got there.

File: pyanaconda/constants.py (hunk 5 above)
Comment (jkonecny12) [inline pyanaconda/constants.py]: I was in temptation to do it with ``Enum`` here, ok I'll change this. :)

File: docs/boot-options.rst (hunk 12 above)
Comment (jkonecny12) [inline docs/boot-options.rst]: Sorry it looks like some leftover. I'll remove that.

**Comment by jkonecny12** [comment]
jenkins, test this please

**Comment by jkonecny12** [comment]
jenkins, test this please

**Comment by jkonecny12** [comment]
jenkins, test this please

**Comment by jkonecny12** [comment]
jenkins, test this please

**Comment by jkonecny12** [comment]
jenkins, test this please

**Comment by jkonecny12** [comment]
Tests are failing but it's not issue of these changes. Ready for next review. @vpodzime, @M4rtinK could you please look on the changes if everything is alright now?

**Comment by jkonecny12** [comment]
Thank you all for these valuable responses :+1:
